Internal: Ally Announcements [ED-21533]

LorandBE opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Add accessibility announcement modal to Elementor editor promoting Ally plugin for website accessibility scanning and remediation.

Main changes:
- Created a11y-announcements module with React-based welcome dialog displaying promotional video and plugin installation CTA
- Added conditional display logic checking user permissions, Ally plugin status, and AI announcement history before showing modal
- Updated Playwright test helpers to automatically dismiss announcement modals preventing test interference

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->

LorandBE committed:
Added the necessary files

LorandBE committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

LorandBE committed:
Lint fixes

LorandBE committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

LorandBE committed:
Design fix

LorandBE committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

LorandBE committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

mareksusniak-coder commented on modules/a11y-announcements/assets/js/welcome-screen.js:
```diff
@@ -0,0 +1,93 @@
+		setIsMounted( true );
+	}, [] );
+
+	if ( ! isMounted || ! anchorElRef.current ) {
```
`!anchorElRef.current ` isn't enough?

mareksusniak-coder commented on modules/a11y-announcements/assets/js/app.js:
```diff
@@ -0,0 +1,74 @@
+			fetch( window.elementorA11yAnnouncement.ajaxUrl, {
+				method: 'POST',
+				body: formData,
+			} ).catch( () => {
```
It's minor, but this notation look weird, maybe `catch(() => {})` in one line can be more clear?

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Update

rami-elementor committed:
Use internal Elementor methods for the announcement

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Update

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Update

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

ManorHazaz committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

rami-elementor committed:
Update

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

Svitlana-Dykun committed:
fix(tests): close a11y announcement modal in test setup

Svitlana-Dykun committed:
fix(tests): close a11y announcement modal in test setup

- Add e-a11y-announcement to closeAnnouncementsIfVisible in WpAdminPage
- Add closeAnnouncementsIfVisible method to EditorPage
- Call closeAnnouncementsIfVisible in waitForPanelToLoad to handle page reloads

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

Svitlana-Dykun committed:
ci: trigger CI rebuild

Svitlana-Dykun committed:
fix(tests): close a11y announcement before publishing page

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement

Svitlana-Dykun committed:
Internal: Fix failed tests for ally (#34091)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Fix flaky Playwright test failures by improving cleanup of
modal dialogs and announcement popups that were interfering with test
execution.

Main changes:
- Added Escape key press and MuiDialog/MuiModal DOM cleanup to
closeAnnouncementsIfVisible() in editor and admin pages
- Implemented ensureNotFavorited() helper method to guarantee clean
state before adding favorite widgets in tests

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before
using.
💡 **Tip:** You can customize your AI Description using **Guidelines**
[Learn
how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->

Svitlana-Dykun committed:
Merge branch 'main' into ED-21533-ally-a11y-editor-anouncement
